src/Core/Framework/Adapter/Twig/Extension/BuildBreadcrumbExtension.php

```diff
@@ -74,17 +98,53 @@ public function getFullBreadcrumb(array $twigContext, CategoryEntity $category,
     }
 
     /**
+     * @deprecated tag:6.8.0 - Parameter $twigContext will be removed, as it is not needed anymore
+     * @deprecated tag:6.8.0 - reason:return-type-change - Will only return `array<string, SalesChannelCategoryEntity>`
+     *
      * @param array<string, mixed> $twigContext
      *
-     * @return array<string, CategoryEntity>
+     * @return array<string, CategoryEntity|SalesChannelCategoryEntity>
      */
-    public function getFullBreadcrumbById(array $twigContext, string $categoryId, Context $context): array
+    public function getFullBreadcrumbById(array $twigContext, string $categoryId, Context|SalesChannelContext $context): array
     {
-        $category = $this->categoryRepository->search(new Criteria([$categoryId]), $context)->getEntities()->first();
-        if (!$category instanceof CategoryEntity) {
+        if (Feature::isActive('v6.8.0.0')) {
```

Comment:
same as above.

Reply:
Argh, I forgot to commit the proper changes :see_no_evil: for the `getFullBreadcrumb` method...

I meant it to be like it is here, i.e. if the Feature `v6.8.0.0` is active, we can assert that the context is an instance of the `SalesChannelContext` and directly use the `salesChannelCategoryRepository`, otherwise we try to determine the proper context, is this okay? Or should I turn the conditions around? :thinking: